tests/e2escenarios/e2e_test.go

```diff
@@ -259,6 +259,230 @@ var _ = Describe("E2E Test", func() {
+			Eventually(string(commonVar.CliRunner.Run(getSVCArgs...).Out.Contents()), 60, 3).ShouldNot(ContainSubstring(serviceName))
+		})
+	})
+	Context("starting with non-empty Directory add Binding", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			commonVar.CliRunner.EnsureOperatorIsInstalled("service-binding-operator")
+			commonVar.CliRunner.EnsureOperatorIsInstalled("cloud-native-postgresql")
+			Eventually(func() string {
+				out, _ := commonVar.CliRunner.GetBindableKinds()
+				return out
+			}, 120, 3).Should(ContainSubstring("Cluster"))
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			addBindableKind := commonVar.CliRunner.Run("apply", "-f", helper.GetExamplePath("manifests", "bindablekind-instance.yaml"))
+			Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		})
+		It("should verify developer workflow of using binding as file in innerloop", func() {
+			deploymentName := "my-component"
+			serviceName := "my-cs"
+			bindingName := helper.RandString(6)
+			getDeployArgs := []string{"get", "deployment", "-n", commonVar.Project}
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				// helper.ExpectString(ctx, "Based on the files in the current directory odo detected")
+				helper.ExpectString(ctx, "Language: javascript")
+				helper.ExpectString(ctx, "Project type: nodejs")
+				helper.ExpectString(ctx, "Is this correct")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			//add binding information (binding as file)
+			helper.Cmd("odo", "add", "binding", "--name", bindingName, "--service", "cluster-sample").ShouldPass()
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, _, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// check odo describe to check for env
+			stdout := helper.Cmd("odo", "describe", "binding", "-o", "json").ShouldPass().Out()
```

Comment:
Maybe testing the human-readable output instead would make more sense, from a developer's perspective?